activerecord/lib/active_record/database_configurations/hash_config.rb

```diff
@@ -114,8 +113,12 @@ def schema_cache_path
         configuration_hash[:schema_cache_path]
       end
 
-      def default_schema_cache_path
-        "db/schema_cache.yml"
+      def default_schema_cache_path(db_dir = "db")
+        if primary?
+          "#{db_dir}/schema_cache.yml"
+        else
+          "#{db_dir}/#{name}_schema_cache.yml"
```

Comment:
```suggestion
          File.join(db_dir, "schema_cache.yml")
        else
          File.join(db_dir, "#{name}_schema_cache.yml")
```

Nowhere near critical, but `db_dir` being user suplied, we don't know if it already have a trailing `/`. Also potentially support `\` on Windows (other things would probably break but hey...)

Reply:
> Also potentially support \ on Windows (other things would probably break but hey...)

I was trying to validate this but it seems Ruby just use `/` on `File.join` on Windowns as well. But agree, I prefer to use it as well here.